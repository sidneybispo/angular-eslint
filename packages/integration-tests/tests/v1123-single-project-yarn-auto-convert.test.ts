// utils/local-registry-process.ts
import { execSync } from 'child_process';

export const runYarnInstall = () => execSync('yarn install');

export const runNgAdd = (packageName: string) =>
  execSync(`ng add ${packageName} --registry=http://localhost:4200/`);

export const runConvertTSLintToESLint = (args: string[]) =>
  execSync(`ng convert --from-to tslint=eslint ${args.join(' ')}`);

// utils/require-uncached.ts
const cache = new Map();

export const requireUncached = <T>(modulePath: string): T => {
  if (cache.has(modulePath)) {
    cache.delete(modulePath);
  }

  const moduleExports = require(modulePath);
  cache.set(modulePath, moduleExports);
  return moduleExports;
};

// utils/run-lint.ts
import { execSync } from 'child_process';

export const runLint = (fixtureDirectory: string) =>
  execSync(`ng lint ${fixtureDirectory}`).toString();

// __tests__/convert-tslint-to-eslint.test.ts
import path from 'path';
import { runY
