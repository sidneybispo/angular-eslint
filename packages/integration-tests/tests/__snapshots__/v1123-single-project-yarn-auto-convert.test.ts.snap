// Jest Snapshot v1, https://goo.gl/fbAQLP

// Configuration for eslint to lint TypeScript and HTML files
exports[`v1123-single-project-yarn-auto-convert it should pass linting after converting the out of the box Angular CLI setup (with a custom prefix set) 1`] = `
{
  "ignorePatterns": [
    "projects/**/*"
  ],
  "overrides": [
    {
      "extends": [
        "plugin:@angular-eslint/ng-cli-compat",
        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "createDefaultProgram": true,
        "project": [
          "tsconfig.json",
          "e2e/tsconfig.json"
        ]
      },
      "rules": {
        "@angular-eslint/component-selector": [
          "error",
          {
            "prefix": "some-custom-prefix",
            "style": "kebab-case",
            "type": "element"
          }
        ],
        "@angular-eslint/directive-selector": [
          "error",
          {
            "prefix": "some-custom-prefix",
            "style": "camelCase",
            "type": "attribute"
          }
        ]
      }
    },
    {
      "extends": [
        "plugin:@angular-eslint/template/recommended"
      ],
      "files": [
        "*.html"
      ],
      "rules": {}
    }
  ],
  "root": true
}
`;

// Configuration for the Angular ESLint builder to lint TypeScript and HTML files
exports[`v1123-single-project-yarn-auto-convert it should pass linting after converting the out of the box Angular CLI setup (with a custom prefix set) 2`] = `
{
  "builder": "@angular-eslint/builder:lint",
  "options": {
    "lintFilePatterns": [
      "src/**/*.ts",
      "src/**/*.html"
    ]
  }
}
`;

// Output of the lint command
exports[`v1123-single-project-yarn-auto-convert it should pass linting after converting the out of the box Angular CLI setup (with a custom prefix set) 3`] = `
"
Linting "v1123-single-project-yarn-auto-convert"...

__ROOT__/v1123-single-project-yarn-auto-convert/src/test.ts
  13:5  error  Member call should be declared before all method definitions  @typescript-eslint/member-ordering

âœ– 1 problem (1 error, 0 warnings)
"
`;
